R/fit.R

```diff
@@ -578,6 +578,11 @@ CmdStanMCMC <- R6::R6Class(
       if (!length(self$output_files(include_failed = FALSE))) {
         stop("No chains finished successfully. Unable to retrieve the draws.")
       }
+      if (inc_warmup && !private$sampling_info_$save_warmup) {
+        stop("Warmup draws were requested from a fit object without them! ",
+             "Please rerun the model with save_warmup = TRUE.", call. = FALSE)
+      }
+
```

Comment:
unrelated to the bug, but we should throw this error immediately instead of at the end of the function after doing the work of reading in the csv